handler.py

```diff
@@ -40,43 +39,62 @@ def on_get(self, request, response, *args, **kwargs):
         params = request.get_params()
         self.logger.debug("on_get, params: {0}".format(params))
 
-        if params.get('identifier', '') == 'refresh':
-            cfg_type = params.get('cfg_type', CfgType.all.name)
+        if params.get('identifier', '') != 'refresh':
+             msg = "Invalid parameters: {0} in 'config': {0}".format(params)
+             self.logger.warning(msg)
+             raise ValueError(msg)
+
+        cfg_type = params.get('cfg_type', CfgType.all.name)
+
+        for current_enum in CfgType:
+            if current_enum.name == cfg_type:
+                self._trigger_config_change_hook(current_enum)
+                return
+        msg = "Invalid 'config' refresh type: {0}".format(cfg_type)
+        self.logger.warning(msg)
+        raise ValueError(msg) 
+
+    def on_put(self, request, response, *args, **kwargs):
+        """ API endpoint for configuration component
+
+        Example:
+            http://[host]:[port]/config/update
+
+        """
+         # Ensure instance "execute" access
+        ensure_access("instance", "execute")
+
+        params = request.get_params()
+        self.logger.debug("on_put, params: {0}".format(params))
+
+        if params.get('identifier', '') != 'update':
+             msg = "Invalid parameters: {0} in 'config': {0}".format(params)
+             self.logger.warning(msg)
+             raise ValueError(msg)
 
-            for current_enum in CfgType:
-                if current_enum.name == cfg_type:
-                    self._trigger_config_change_hook(current_enum)
-                    return
-            msg = "Invalid 'config' refresh type: {0}".format(cfg_type)
-            self.logger.warning(msg)
-            raise ValueError(msg)
-        elif params.get('identifier', '') == 'update':
+        token = params.get('token', self._instance_api_key)
 
-            token = params.get('token', None)
-            # allow to override gathered settings
-            url_prefix = params.get('url', self._product_api_url_prefix)
-            instance_id = params.get('instance_id', self._instance_id)
+        # allow to override gathered settings
+        url_prefix = params.get('url', self._product_api_url_prefix)
+        instance_configuration_id = params.get('instance_configuration_id', None)
+        instance_configuration_version_id =\
+            params.get('instance_configuration_version_id', None)
 
-            url = "{}/instances/{}/config".format(url_prefix, instance_id)
+        org_id = params.get('nio-organization', None)
```

Comment:
I would think we would want to update the API if using `api_key` authentication to not need to have a `nio-organziation` passed. We can get that from the `api_key` and once we have the polling mechanism, it would be unfortunate to have to hardcode organization_ids onto physical devices.

Reply:
Agreed, a given API key can really only belong to one organization